noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/common.nr

```diff
@@ -206,7 +206,7 @@ pub fn update_public_end_values(public_call: PublicCallData, circuit_outputs: &m
     propagate_new_note_hashes(public_call, circuit_outputs);
 
     propagate_new_l2_to_l1_messages(public_call, circuit_outputs);
-
+    propagate_new_unencrypted_logs(public_call, circuit_outputs);
```

Comment:
Would it be best to have a flag for each `propagate` function to tell it whether to push to `end` or `end_non_revertible`? Rather than having 2 of each function!
So in public `setup` and `teardown`, when calling `update_public_end_non_revertible_values`, we should push the logs to `.end_non_revertible` rather than `.end`? (Previously, the single hash was just accumulated to `end` in all cases)

Reply:
For now I added a new function, mimicing the rest of the code, just to avoid a lot of changes. I'd be happy to take on the task of reducing the code here, just in another PR?